### PR TITLE
First draft of lab

### PR DESCRIPTION
**Pull request by mdippery**
I came up with what I think is a fun project for this lab. :-) My only worry is that maybe it's a bit _too_ challenging? When I started it off, it actually seemed too _easy_, but as I worked through it I found that maybe it's a bit harder than I expected. Let me know what you think and I can adjust it accordingly (or even scrap it entirely if need be).

cc: @JimCampagno @AnnJohn

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,16 +18,64 @@
...
+2. A segmented control that allows the user to select a color.
+
+When the user selects a different color, the light bulb's hue should change. Easy, right? Here's what it looks like when the user presses **Green**:
+
+![LightBulb: Green Edition](http://i.imgur.com/m5aboWl.png)
+
+### Using Code
+
+Part of being a competent Swift programmer is being able to use and modify an existing code base to accomplish your task. To reinforce this idea, part of this app has been provided for you. Open up `LightBulb/LightBulb.xcodeproj` in Xcode to see what has been given to you.
+
```
Comment (JimCampagno) [inline README.md:47]: This is music to my ears... perfect.

File: README.md [hunk 2]
```diff
@@ -20,16 +18,64 @@
...
-* Maybe it's something like this [lab](https://github.com/learn-co-curriculum/swift-viewLifeCycle-lab) but not as difficult.  
-* Free to explore ideas or fun projects the student can take on at this point.
+If you complete the following steps, you should end up with a properly working app:
 
+1. Hook up the `lightBulb` IB outlet in `ViewController.swift` with the light bulb view in `Main.storyboard`.
+2. Add a segmented control underneath the lightbulb.
+	* By default, a segmented control only has 2 buttons. Yours needs to have 4. Do you know how you can change that? (If you guessed "in the Attributes Inspector", you'd be correct!)
+	* The individual parts of a segmented control also have default names. Try to figure out on your own how you can change those names. (Hint: Look for a section in the Attributes Inspector called **Segment**. It is a popup menu that should list all the parts of your segmented control. You'll want to change an attribute called **Title** for each segment part.)
+3. Hook up the segmented control to the `colorSelected(_:)` IB outlet in `ViewController.swift`.
```
Comment (JimCampagno) [inline README.md]: Should this read IBAction? I'm cool with leaving IB outlet if that's a way we can word it, I'm not sure.

Side note, considering I'm making comments throughout that read like... "Perfect" or... "Love this", do you want me to tag you in the actionable ones or the ones that require you to respond like so --> @mdippery

File: README.md [hunk 3]
```diff
@@ -20,16 +18,64 @@
...
+4. Modify the `colorSelected(_:)` method where shown. It should change the background color the `lightBulb` view to the chosen color.
```
Comment (JimCampagno) [inline README.md]: ~~Should read as:~~

~~background color of the `lightBulb` view~~

~~WILL ADJUST THIS MYSELF.~~

File: README.md [hunk 4]
```diff
@@ -20,16 +18,64 @@
...
+	* Doing this might be a bit tricky. Notice that there is a `print` statement above the part of this method that you need to modify. That `print` statement is a hint as to how you can figure out which segment the user selected. Remember: You can get help using Xcode's documentation.
```
Comment (JimCampagno) [inline README.md:72]: I don't see a print function here. Did you mean to take it out?

File: README.md [hunk 5]
```diff
@@ -20,16 +18,64 @@
...
+	* The selected part of the segmented control will be returned as an `Int`. Remember: You have already been provided with a method that can turn an `Int` into a color!
+5. Modify the `changeColor(_:)` method to change the background color of the `lightBulb` view.
```
Comment (JimCampagno) [inline README.md]: ~~Note to self, change `changeColor(_:)` to `changeColor(to:)` considering we provided an external parameter to the first argument.~~

File: README.md [hunk 6]
```diff
@@ -20,16 +18,64 @@
...
+6. Finally, modify `viewDidLoad()` to set the initial color of the light bulb to red.
+	* Remember: You have two methods to help you. One is `colorFromString(_:)`, which can take a string like "red" or "blue" and return a color.
```
Comment (JimCampagno) [inline README.md:76]: I don't see this `colorFromString(_:)` method. I'm down for it to be included, if you do wind up leaving it out I think then we should show them how to work with your extension with an example like this:

``` swift
view.backgroundColor = "blue".color
```

File: LightBulb/LightBulb/ViewController.swift [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+//
+
+import UIKit
+
+class ViewController: UIViewController {
+    @IBOutlet weak var lightBulb: UIView!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // TODO: Change background color to red
```
Comment (JimCampagno) [inline LightBulb/LightBulb/ViewController.swift]: All the TODO's, I think we should add `lightBulb` view... as in..

``` swift
// TODO: Change background color of lightBulb view to red
```

That way, they don't jump right to the TODO's and think they should change the background color of the view.

**Comment by JimCampagno** [comment]
I really **really** like this lab. I love the extensions on `String` and `UISegmentedControl`, clever stuff.

Some small changes needed to made per my notes above, some of them I'm going to do myself right now. I also want to discuss a possible new workflow with you considering how we're definitely on the same page. Unless I feel there's a change that needs to be made on your end which will help you see how we like to do things, or a style I like to present it in... I'm just going to make note of the change and do it myself so that way you have more time to concentrate on the labs ahead.

I think we can add unit tests to this lab. Right now, we're holding a discussion on whether or not to use Xcode 10 with Swift 3.0, if so.. then we would have to use XCTest and not Quick & Nimble (as originally intended). That being said, I'm going to add unit tests to this lab myself and become familiar with XCTest in the process.

@mdippery

File: README.md (hunk 2 above)
Comment (mdippery) [inline README.md]: Fixed!

I read through all the comments, so you can keep doing it the way you are. If I happen to miss one that I need to take action on, though, feel free to nudge me.

File: LightBulb/LightBulb/ViewController.swift (hunk 7 above)
Comment (mdippery) [inline LightBulb/LightBulb/ViewController.swift]: Good point, fixed.

**Comment by mdippery** [comment]
Excellent. I'll take a look at the unit test stuff, too, to get a feel for how we'd like that to be done.

**Pull request merged**